docs/templates/properties.js

```diff
@@ -39,6 +39,21 @@ module.exports['has-errors'] = function ({ item }) {
   return item.inheritance.some(c => c.errors.length > 0);
 };
 
+module.exports.functions = function ({ item }) {
+  return [
+    ...[...findAll('FunctionDefinition', item)].filter(f => f.visibility !== 'private'),
+    ...[...findAll('VariableDeclaration', item)].filter(f => f.visibility === 'public'),
+  ];
```

Comment:
I think this works:

```suggestion
  return [
    ...findAll('FunctionDefinition', item).filter(f => f.visibility !== 'private'),
    ...findAll('VariableDeclaration', item).filter(f => f.visibility === 'public'),
  ];
```

Reply:
Also, we can keep the constructor getters at first

```suggestion
  const functionDefs = findAll('FunctionDefinition', item)
  const constructor = functionDefs.find(f => f.kind == 'constructor');
  return [
    constructor,
    ...findAll('VariableDeclaration', item).filter(f => f.visibility === 'public'),
    ...functionDefs.filter(f => f.visibility !== 'private')
  ];
```

For cases like AccessManager and AccessControl I think it's valuable to have the `ADMIN_ROLE`/`DEFAULT_ADMIN_ROLE` definitions at first. 

It's true that `ADMIN_ROLE` and `PUBLIC_ROLE` don't have NatSpec anyways, so I wouldn't have problems keeping the current code.

Reply:
>I think this works

It doesn't because `findAll` is a generator function, it doesn't return an array.